OG image cropping

Newnab opened the pull request:
If appsettings contains relevant crop sizes, force any og image to be the best possible dimensions. Adjacent PR incoming to add the sizes to our site boilerplate.

AdamBurtEtch committed:
Whitespace formatting

AdamBurtEtch committed:
Add crop parameters to og image

If appsettings contains relevant crop sizes, force any og image to be the best possible dimensions.

peterkeating commented on MetaTags/Services/MetaTagsService.cs:
```diff
@@ -124,13 +124,13 @@ private string GetHostUrl()
 
             var imageUrl = _mediaFileStore.MapPathToPublicUrl(path);
-            return imageUrl.StartsWith("http") ? imageUrl : $"{GetHostUrl()}{imageUrl}";
+            return imageUrl.StartsWith("http") ? imageUrl : $"{GetHostUrl()}{imageUrl}?width=1200&height=630&rmode=crop";
```
We'll need to make sure that `630` is added to the app settings in the site boilerplate.

https://github.com/EtchUK/Etch.OrchardCore.SiteBoilerplate/blob/main/Content/src/Etch.OrchardCore.SiteBoilerplate/appsettings.json

peterkeating replied:
Also not a fan of 1200 & 630 being magic numbers, can we move them to constants.

Newnab replied:
Committed some changes. Is that what you meant?

peterkeating replied:
Not quite. You need to add the following to the top of the class.

```
private const int ImageCropHeight = 630;
private const int ImageCropWidth = 1200;
```

Then use those variables 👍

AdamBurtEtch committed:
Move width and height into constants

AdamBurtEtch committed:
Amend constants for image crop sizes

AdamBurtEtch committed:
Update meta image hint

To reflect the new status quo in terms of recommended size, and also that the site can crop accordingly if a big enough image is provided.
